lesson05/circle_test.go

```diff
@@ -0,0 +1,108 @@
+package main
+
+import (
+	"math"
+	"reflect"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestAreaC(t *testing.T) {
+
+	tests := []struct {
+		name    string
+		args    Circle
+		want    float64
+		wantErr bool
+	}{
+		{
+			name: "if radius is <= 0 should return error ",
+			args: Circle{
+				Radius: 0.0,
+			},
+			want:    0.0,
+			wantErr: true,
+		},
+		{
+			name: "if radius = 1 should return Pi",
+			args: Circle{
+				Radius: 1.0,
+			},
+			want:    math.Pi,
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := (tt.args).Area()
+
+			if (err != nil) != tt.wantErr {
+				t.Errorf("Area() error = %v, wantErr %v, got = %v", err, tt.wantErr, got)
+				assert.Equal(t, err, tt.wantErr)
+				return
```

Comment:
The test should not return, but continue to test until the function end. basically you could also test if the function returns 0.0, which it should. Right now, after testing the first case for error it will exit and skip this assertion of the value `got`